### PR TITLE
[CosmosDB] Generate SDK for 2020-06-01-preview version

### PR DESCRIPTION
**Pull request by kavskalyan**
Includes changes related to PITR and backup policy

### CODE REVIEW

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure CosmosDB Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure CosmosDB Resources.")]
 
-[assembly: AssemblyVersion("1.2.0.0")]
-[assembly: AssemblyFileVersion("1.2.0.0")]
+[assembly: AssemblyVersion("1.3.0.0")]
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Properties/AssemblyInfo.cs:10]: @erich-wang is there a way we can incorporate the preview tag in Assembly Version

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/RestoreDatabaseAccountOperationsTests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    },
+                    Kind = "GlobalDocumentDB",
+                    Properties = databaseAccountCreateUpdateProperties
+                };
+
+                DatabaseAccountGetResults restoredDatabaseAccount = (await cosmosDBManagementClient.DatabaseAccounts.CreateOrUpdateWithHttpMessagesAsync(resourceGroupName, restoredatabaseAccountName, databaseAccountCreateUpdateParameters)).Body;
+                Assert.NotNull(restoredDatabaseAccount);
+                Assert.NotNull(restoredDatabaseAccount.RestoreParameters);
+                Assert.Equal(restoredDatabaseAccount.RestoreParameters.RestoreSource.ToLower(), restorableDatabaseAccount.Id.ToLower());
+                Assert.True(restoredDatabaseAccount.BackupPolicy is ContinuousModeBackupPolicy);
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/RestoreDatabaseAccountOperationsTests.cs:76]: Is there a reason you not using Assert.Equal ?

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj [hunk 3]
```diff
@@ -6,17 +6,17 @@
...
     <PackageReleaseNotes>
       <![CDATA[
-        Updating with the apiVersion 2020-04-01 azure-rest-api-specs,
+        Updating with the apiVersion 2020-06-01-preview azure-rest-api-specs,
         Includes: 
-        Updating the CreateUpdateOptions by removing additionalProperties, changing the type of Throughput from string to int, adding AutoscaleSettings. 
-        ProvisionedThroughputSettings is renamed to AutoscaleSettings in ThroughputSettingsResource.
-        Replacing IpRangeFilter by IpRules, which is a list of IpAddressOrRange objects.
-        Introducing EnableFreeTier and ApiProperties in DatabaseAccount properties. 
+		Seperating the DatabaseAccountCreateUpdateProperties from DatabaseAccountCreateUpdateParameters
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj:16]: nit: "Separating"

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj [hunk 4]
```diff
@@ -6,17 +6,17 @@
...
     <Description>Provides developers with libraries for the CosmosDB under Azure Resource manager to manage CosmosDB Account, Databases and child resources and their available management capabilities. Create, Delete, Update CosmosDB Account, Databases and child resources. Note: This client library is for CosmosDB under Azure Resource Manager.</Description>
-    <Version>1.2.0</Version>
+    <Version>1.3.0-preview</Version>
     <AssemblyName>Microsoft.Azure.Management.CosmosDB</AssemblyName>
     <PackageTags>management;cosmosdb;</PackageTags>
     <PackageReleaseNotes>
       <![CDATA[
-        Updating with the apiVersion 2020-04-01 azure-rest-api-specs,
+        Updating with the apiVersion 2020-06-01-preview azure-rest-api-specs,
         Includes:
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj:15]: Can you elaborate  the changes a little more? Explain what is included in DatabaseAccountCreateUpdateProperties. Sum all the changes related to PITR in one para. Please mention, this is for the private preview of PITR.

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/RestoreDatabaseAccountOperationsTests.cs (hunk 2 above)
Comment (kavskalyan) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/RestoreDatabaseAccountOperationsTests.cs:76]: The backupPolicy is the base class.
For accounts with Periodic mode backup type, the derived class "PeriodicModeBackupPolicy" is returned. 
For accounts with Continuous mode backup type, the derived class "ContinuousModeBackupPolicy" is returned. 
So to assert if the restored account has continuous backup policy, we just need to check if backupPolicy is the ContinuousModeBackupPolicy,

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj (hunk 3 above)
Comment (kavskalyan) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj]: Fixed it

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj (hunk 4 above)
Comment (kavskalyan) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj:16]: Elaborated the changes now, Please let me know if you would like me to expand more.

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (isra-fel) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Properties/AssemblyInfo.cs:10]: No. Aseembly version are just numbers

**Review by MehaKaushik — APPROVED** [review]
LGTM

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**